library/src/main/java/in/krharsh17/barview/BarView.java

```diff
@@ -298,6 +298,23 @@ public static String getRandomColor() {
         return color.toString();
     }
 
+    /**
+     * This function returns a random color based on a constant {@value #CHAR_ARRAY}
+     * It modifies the passed in hex value by randomly choosing between any of the 16
+     * literals in the char array and replacing them from index 4 till 6 in the passed in
+     * literal of the format #XXXXXX
+     *
+     * @param preciseColor is the color user wants the bar to be precisely like
+     * @return
+     */
+    public static String getRandomColor(StringBuilder preciseColor) {
+        char[] letters = CHAR_ARRAY.toCharArray();
+        for (int i = 4; i < 7; i++) {
+            preciseColor.setCharAt(i,letters[(int) Math.round(Math.floor(Math.random() * 16))]);
+        }
+        return preciseColor.toString();
```

Comment:
Could you provide a few sample input + output examples for this? Run this on a separate java environment, and attach the list of input colours and the corresponding list of output colours, please.

Reply:
I would do that after fixing above things up